src/coreclr/inc/corinfo.h

```diff
@@ -1729,8 +1730,11 @@ struct CORINFO_THREAD_STATIC_BLOCKS_INFO
 {
     CORINFO_CONST_LOOKUP tlsIndex;
     uint32_t offsetOfThreadLocalStoragePointer;
-    uint32_t offsetOfMaxThreadStaticBlocks;
-    uint32_t offsetOfThreadStaticBlocks;
+    uint32_t offsetOfNonGCMaxThreadStaticBlocks;
```

Comment:
Instead of adding both GC and non-GC flavors of all offsets, could you please change `getThreadLocalStaticBlocksInfo` to take the field handle as an argument and let runtime decide which set of offsets is applicable to the field and just return those?

Reply:
It should be also merged with `get{Non}GCThreadLocalFieldInfo` into a single method `getThreadStaticFieldInfo` that will returns all information you need to generate inline access for the given thread static field.

A common pattern in the JIT/EE interface design is to have methods that return "everything that the JIT needs to know to do something". `getCallInfo` is a good example of such method. This should be another one of the methods that follow this pattern.

Reply:
Currently, the importer has the field handle and I can use that to combine `getNonGCThreadLocalFieldInfo` and `getGCThreadLocalFieldInfo`. However, the helper expansion happens in later phase (when field handle is not available) and that's when I need the information from `getThreadStaticFieldInfo()`.  All I need is if it is a non-gc or gc-type and I already have that information. So I can combine `getNonGCThreadLocalStaticBlocksInfo()` and `getGCThreadLocalStaticBlocksInfo()` and it will return the appropriate offsets depending on if I am pollling it for gc or non-gc type. 

However, I don't see a way to combine `getThreadLocalFieldInfo()` and `getThreadLocalStaticBlocksInfo()`. If I have to, and get all this information during importer, I will have to keep some kind of dictionary where importer will save the offset values against the `GenTreeCall*` node. In later phase, I will get the contents from dictionary for the corresponding `GenTreeCall*` and do the expansion. It makes it complicated than needed, so probably having `getThreadLocalFieldInfo()` and `getThreadLocalStaticBlocksInfo()` should be enough?

Reply:
Ok, I think it would be a good idea to combine the GC and non-GC flavors for now at least.

We can keep `getThreadLocalFieldInfo()` and `getThreadLocalStaticBlocksInfo()` separate for now. I am not sure whether we will be able to keep it that way for AOT, but we can deal with it once/if it becomes actual problem.